modules/lib.lua

```diff
@@ -257,12 +257,18 @@ if isServer then
         while not DoesEntityExist(veh) do Wait(0) end
         while GetVehicleNumberPlateText(veh) == '' do Wait(0) end
 
+        Entity(veh).state:set('initVehicle', true, true)
+        Entity(veh).state:set('setVehicleProperties', props, true)
+
+        lib.waitFor(function()
+            local state = Entity(veh).state
```

Comment:
cache the state outside the waitFor and use your cached state for setting the state bags as well

Reply:
Would it still update the value of the statebag? We are trying to see if it cleared.

Reply:
Yes, this should be a reference to the statebag that's localized, I believe the memory location is shared.